library/ssl_srv.c

```diff
@@ -1872,27 +1872,43 @@ static int ssl_parse_client_hello( mbedtls_ssl_context *ssl )
     got_common_suite = 0;
     ciphersuites = mbedtls_ssl_get_protocol_version_ciphersuites( ssl->conf, ssl->minor_ver );
     ciphersuite_info = NULL;
-#if defined(MBEDTLS_SSL_SRV_RESPECT_CLIENT_PREFERENCE)
-    for( j = 0, p = buf + ciph_offset + 2; j < ciph_len; j += 2, p += 2 )
-        for( i = 0; ciphersuites[i] != 0; i++ )
-#else
-    for( i = 0; ciphersuites[i] != 0; i++ )
+
+    if (ssl->conf->respect_cli_pref == MBEDTLS_SSL_SRV_RESPECT_CLIENT_PREF_ENABLED)
+    {
         for( j = 0, p = buf + ciph_offset + 2; j < ciph_len; j += 2, p += 2 )
-#endif
-        {
-            if( p[0] != ( ( ciphersuites[i] >> 8 ) & 0xFF ) ||
-                p[1] != ( ( ciphersuites[i]      ) & 0xFF ) )
-                continue;
+            for( i = 0; ciphersuites[i] != 0; i++ )
+            {
+                if( p[0] != ( ( ciphersuites[i] >> 8 ) & 0xFF ) ||
```

Comment:
Please avoid duplicating code. It would be better to create an auxiliary function for the loop body. (I'd prefer an auxiliary function for the double loop, but because the two sides don't encode the ciphersuite list in the same way, this is not possible without extra complications.)

Reply:
Yes, I tried to find a solution with an aux function, but the code became complicated and unclear and finally I decided to leave it as it is. Do you want me to work on it?

Reply:
No, if it's too complicated, never mind.